wgsl/index.bs

```diff
@@ -4388,16 +4544,54 @@ value in one type as a value in another type.
     <td>Identity transform. [=Component-wise=] when |T| is a vector.<br>
     The result is |e|.
 
-  <tr algorithm="general reinterpretation">
+  <tr algorithm="32-bit scalar reinterpretation">
     <td>|e|: |T1|<br>
-    |T1| is a [=numeric scalar=] or [=numeric vector=] type<br>
-    |T2| is not |T1| and is a numeric scalar type if |T1| is a scalar, or<br>
-    a numeric vector type if |T1| is a vector
+    |T1| is i32, u32, or f32<br>
+    |T2| is not |T1| and is i32, u32, or f32
     <td class="nowrap">bitcast&lt;|T2|&gt;(|e|): |T2|
-    <td>Reinterpretation of bits as |T2|. [=Component-wise=] when |T1| is a vector.<br>
+    <td>Reinterpretation of bits as |T2|. <br>
     The result is the reinterpretation of the bits in |e| as a |T2| value.
+
+  <tr algorithm="32-bit elements vector reinterpretation">
+    <td>|e|: vec|N|&lt;|T1|&gt;<br>
+    |T1| is i32, u32, or f32<br>
+    |T2| is not |T1| and is i32, u32, or f32
+    <td class="nowrap">bitcast&lt;vec|N|&lt;|T2|&gt;&gt;(|e|): vec|N|&lt;|T2|&gt;
+    <td>[=Component-wise=] reinterpretation of bits as |T2|.<br>
+    The result is the reinterpretation of the bits in |e| as a vec|N|&lt;|T2|&gt; value.
+
+  <tr algorithm="binary16x2 to 32-bit scalar reinterpretation">
+    <td>|e|: vec2&lt;f16&gt;<br>
+    |T| is i32, u32, or f32
+    <td class="nowrap">bitcast&lt;|T|&gt;(|e|): |T|
+    <td>Reinterpretation of bits as |T|.<br>
+    The result is the reinterpretation of the 32 bits in |e| as a |T| value, following the internal layout rules.
+
+  <tr algorithm="32-bit scalar to binary16x2 reinterpretation">
+    <td>|e|: |T|<br>
+    |T| is i32, u32, or f32
+    <td class="nowrap">bitcast&lt;vec2&lt;f16&gt;&gt;(|e|): vec2&lt;f16&gt;
+    <td>Reinterpretation of bits as vec2&lt;f16&gt;.<br>
+    The result is the reinterpretation of the 32 bits in |e| as a vec2&lt;f16&gt; value, following the internal layout rules.
+
+  <tr algorithm="binary16x4 to vec2 of 32-bit scalar reinterpretation">
+    <td>|e|: vec4&lt;f16&gt;<br>
+    |T| is i32, u32, or f32
+    <td class="nowrap">bitcast&lt;vec2&lt;|T|&gt;&gt;(|e|): vec2&lt;|T|&gt;
+    <td>Reinterpretation of bits as vec2&lt;|T|&gt;.<br>
+    The result is the reinterpretation of the 64 bits in |e| as a vec2&lt;|T|&gt; value, following the internal layout rules.
+
+  <tr algorithm="vec2 of 32-bit scalar to binary16x4 reinterpretation">
+    <td>|e|: vec2&lt;|T|&gt;<br>
+    |T| is i32, u32, or f32
+    <td class="nowrap">bitcast&lt;vec4&lt;f16&gt;&gt;(|e|): vec4&lt;f16&gt;
+    <td>Reinterpretation of bits as vec4&lt;f16&gt;.<br>
+    The result is the reinterpretation of the 64 bits in |e| as a vec4&lt;f16&gt; value, following the internal layout rules.
```

Comment:
This is a really nice way to say it: "following the internal layout rules".

Reply:
To be clear, I'm not asking for a change.  But I think after landing there may be a way to simplify this section by referencing the layout rules for the whole section.